operator/controllers/httpscaledobject_controller.go

```diff
@@ -117,57 +114,33 @@ func (rec *HTTPScaledObjectReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 		return ctrl.Result{}, err
 	}
 
-	// initializes the required variables and set the initial status to unknown
-	httpso.Status = httpv1alpha1.HTTPScaledObjectStatus{
-		ServiceStatus:        httpv1alpha1.Unknown,
-		DeploymentStatus:     httpv1alpha1.Unknown,
-		ScaledObjectStatus:   httpv1alpha1.Unknown,
-		ExternalScalerStatus: httpv1alpha1.Unknown,
-		InterceptorStatus:    httpv1alpha1.Unknown,
-		Ready:                false,
-	}
-	updateStatus(ctx, logger, rec.Client, httpso)
-
 	// httpso is updated now
-
 	logger.Info("Reconciling HTTPScaledObject", "Namespace", req.Namespace, "App Name", appName, "image", image, "port", port)
 
 	// Create required app objects for the application defined by the CRD
 	if err := rec.createOrUpdateApplicationResources(
 		logger,
 		appInfo,
 		httpso,
-		func(httpso *v1alpha1.HTTPScaledObject) {
-			updateStatus(ctx, logger, rec.Client, httpso)
-		},
 	); err != nil {
 		// if we failed to create app resources, remove what we've created and exit
-		logger.Error(err, "Adding app resources")
+		logger.Error(err, "Removing app resources")
 		if removeErr := rec.removeApplicationResources(
 			logger,
 			appInfo,
-			httpso,
-			func(httpso *v1alpha1.HTTPScaledObject) {
-				updateStatus(ctx, logger, rec.Client, httpso)
-			}); removeErr != nil {
+			httpso); removeErr != nil {
 			logger.Error(removeErr, "Removing previously created resources")
 		}
 
 		return ctrl.Result{}, err
 	}
 
-	// If all goes well, set the ready status to true
-	if allReady(httpso) {
-		httpso.Status.Ready = true
-	}
-	updateStatus(ctx, logger, rec.Client, httpso)
```

Comment:
where do we update the status to complete?

Reply:
We don't have a condition to set it as complete, I'll post a comment with an idea here, see what you think of it

Reply:
Added on 1c83ed0